pkg/virt-controller/services/template.go

```diff
@@ -568,7 +568,8 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 
 	}
 	if featuregates.CPUNodeDiscoveryEnabled() {
-		if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Model != "" {
+		if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Model != "" &&
+			vmi.Spec.Domain.CPU.Model != "host-model" && vmi.Spec.Domain.CPU.Model != "host-passthrough" {
```

Comment:
do we want to use constants for these?

Reply:
minor/style: I'd put the host-model check in a different if, or even in a helper function, to make this one simpler to read and to stress that this is a separate check.

Reply:
I would actually encode cpu passthrough differently, a separate boolean maybe. Because we might want to request a machine with certain or better cpu type and then use host-passthrough. But I think it is out of the scope of this patch too.